### PR TITLE
Update `ubuntu-latest` images: set them to Ubuntu 22.04 XOR comment they are `latest` intentionally

### PR DESCRIPTION
**Pull request by konrad-jamrozik**
This is a companion PR to:
- #4996

In this PR, I migrate occurrences of `ubuntu-latest` to Ubuntu 22.04 XOR I add a comment explaining they are intentionally set to `latest`.

I do not have enough context to guess which images should be updated - this is just my best guess. Thus I kindly ask reviewers to verify if my guess is correct.

This PR has revealed following issue:
- #5102

### CODE REVIEW

File: tools/sdk-testgen/ci.yml [hunk 1]
```diff
@@ -1,126 +1,127 @@
...
+    include:
+      - main
+      - feature/*
+      - release/*
+      - hotfix/*
+  paths:
+    include:
+      - tools/sdk-testgen
+
+pool:
```
Comment (konrad-jamrozik) [inline tools/sdk-testgen/ci.yml:28]: This pool is the actual change. Not sure why entire file shows as diffed, perhaps due to automatic line endings conversion.

File: tools/sdk-testgen/ci.yml [hunk 2]
```diff
@@ -1,126 +1,127 @@
...
+  name: azsdk-pool-mms-ubuntu-2204-general
+  vmImage: MMSUbuntu22.04
```
Comment (konrad-jamrozik) [inline tools/sdk-testgen/ci.yml:30]: This pool is the actual change. Not sure why entire file shows as diffed, perhaps due to automatic line endings conversion.

File: tools/js-sdk-release-tools/ci.yml [hunk 3]
```diff
@@ -46,7 +46,7 @@ stages:
         strategy:
           matrix:
             linux:
-              imageName: 'ubuntu-latest'
+              imageName: ubuntu-22.04 # This image is intentionally set to "latest", and not to a specific version
```
Comment (konrad-jamrozik) [inline tools/js-sdk-release-tools/ci.yml:49]:
```suggestion
              imageName: ubuntu-22.04
```

**Review by weshaggard — APPROVED** [review]

**Comment by konrad-jamrozik** [comment]
@weshaggard I am going to override check-enforcer and merge this PR in spite of `tools - mock-service-host - ci` failing. I looked into it and I believe the failure is unrelated to this change. I filed an issue with my findings:
- #5102

**Pull request merged**